### PR TITLE
Add encrypt resource

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    $public_fields['encrypt'] = array(
+      'callback' => array($this, 'getEncrypt'),
+    );
+
+    return $public_fields;
+  }
+
+  /**
+   * @param $value
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: doc.

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+   * @return string
+   */
+  protected function getEncrypt($value) {
+    $privateKey = $value->field_private_key->value();
+
+    if (empty($privateKey)) {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: remove.

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      throw new Exception('CI Build should contain valid Private Key');
+    }
+
+    $request = $this->getRequest();
+    $keyToConvert = array_key_exists('key', $request) ? $request['key'] : NULL;
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: `!empty()`

``` php
if (empty($request['key']) {
  throw new \RestfulBadRequestException
}
```

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    $valueToConvert = array_key_exists('value', $request) ? $request['value'] : NULL;
+
+    if (empty($keyToConvert) || empty($valueToConvert)) {
+      throw new Exception('Request should contain key and value keys.');
+    }
+
+    $nodeServerUrl = variable_get('shoov_ui_build_pr_server', 'http://localhost:3000');
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: $this->nodeJsServer

$node_js_server;

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    $url = $nodeServerUrl . '/encrypt';
+    $options = [
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: [ => array(

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      'method' => 'POST',
+      'data' => "privateKey={$privateKey}&keyToConvert={$keyToConvert}&valueToConvert={$valueToConvert}",
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]:
``` php
'data' => http_build_query(array(
));
```

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      'headers' => array('Content-Type' => 'application/x-www-form-urlencoded'),
+    ];
+
+    $response = drupal_http_request($url, $options);
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php:77]: handle response failure.

``` php
if ($response->code !== 200) {
  throw new \RestfulServerConfigurationException
}
```

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      \RestfulInterface::POST => 'viewEntities',
+    ),
+  );
+
+
+  /**
+   * Overrides \RestfulBase::publicFieldsInfo().
+   */
+  public function publicFieldsInfo() {
+    $public_fields = parent::publicFieldsInfo();
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: remove

**Comment by solker** [comment]
@amitaibu ready for code review.

selection_093

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  public function publicFieldsInfo() {
+    $public_fields['encrypt'] = array(
+      'callback' => array($this, 'getEncrypt'),
+    );
+
+    return $public_fields;
+  }
+
+  /**
+   * Function it is a simple proxy between client request to encode a params and
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: One liner

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+   * nodejs server who have encrypting tool to do that.
+   *
+   * @param $value
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: `@param \EntityMetadataWrapper $wrapper`

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+   *   CI Build object in wrapper.
+   * @return string
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php:40]: line break

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+   *   Encrypted string.
+   */
+  protected function getEncrypt($value) {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php:49]:
``` php
@throws \RestfulBadRequestException
@throws \ RestfulServerConfigurationException
```

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+   */
+  protected $controllers = array(
+    '^.*$' => array(
+      \RestfulInterface::POST => 'viewEntities',
+    ),
+  );
+
+
+  /**
+   * Overrides \RestfulBase::publicFieldsInfo().
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: RestfulBase  => RestfulEntityBase

File: shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * @file
+ * Contains ShoovEncryptResource.
+ */
+
+class ShoovEncryptResource extends \RestfulEntityBase {
+
+  /**
+   * Overrides \RestfulBase::controllers.
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/encrypt/ShoovEncryptResource.class.php]: RestfulBase => ...

**Comment by solker** [comment]
@amitaibu merge please.

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**